packages/synapse-react-client/src/components/OAuthClientManagement/CreateOAuthClient.tsx

```diff
@@ -106,31 +101,37 @@ export const CreateOAuthModal: React.FunctionComponent<
   useEffect(() => {
     setClientName(client?.client_name ?? '')
     setRedirectUris(
-      client?.redirect_uris.map(str => ({ uri: str })) ?? [{ uri: '' }],
+      client?.redirect_uris?.map((str: string) => ({ uri: str })) ?? [
+        { uri: '' },
+      ],
     )
     setPolicyUri(client?.policy_uri ?? '')
     setClientUri(client?.client_uri ?? '')
     setSectorUri(client?.sector_identifier_uri ?? '')
     setUserinfoSignedResponseAlg(
-      client?.userinfo_signed_response_alg ??
-        defaultUserInfoSignedResponseAlgorithm,
+      (client?.userinfo_signed_response_alg ??
+        defaultUserInfoSignedResponseAlgorithm) as
+        | 'JSON'
+        | OIDCSigningAlgorithm,
     )
     setTosUri(client?.tos_uri ?? '')
   }, [isShowingModal, client])
 
   useDebouncedEffect(
     () => {
-      if (accessToken && oAuthClient.client_id) {
+      if (oAuthClient.client_id) {
         // SWC-6365: use the pre-check service to determine if we need to show a warning on edit
-        SynapseClient.isOAuthClientReverificationRequired(
-          oAuthClient,
-          accessToken,
-        ).then(precheckResult => {
-          setWarnTrigger(precheckResult.reverificationRequired)
-        })
+        synapseClient.openIDConnectServicesClient
+          .putAuthV1Oauth2ClientIdVerificationPrecheck({
+            id: oAuthClient.client_id!,
+            oAuthClient: oAuthClient,
+          })
+          .then(precheckResult => {
+            setWarnTrigger(precheckResult.reverificationRequired!)
```

Comment:
`reverificationRequired` is another property that is never null, but because the spec doesn't indicate it as required, we have to add the `!` non-null assertion